src/http.c

```diff
@@ -956,6 +1006,61 @@ http_file_rewind(struct http_file *file)
 	file->offset = 0;
 }
 
+struct http_cookie*
+http_response_cookie(struct http_request *req, char *name, char *val, u_int16_t flags)
+{
+	if (name == NULL || strlen(name) == 0 ||
+		val == NULL || strlen(val) == 0) {
+		kore_log(LOG_ERR, "invalid cookie values to set");
+		return NULL;
+	}
+
+	struct http_cookie	*ck;
+	ck = kore_pool_get(&http_cookie_pool);
+
+	ck->name = kore_strdup(name);
+	ck->value = kore_strdup(val);
+	ck->flags = flags;
+	ck->path = NULL;
+	ck->domain = NULL;
+	ck->expires = 0;
+	ck->maxage = UINT32_MAX;
+
+	TAILQ_INSERT_TAIL(&(req->resp_cookies), ck, list);
+	return ck;
+}
+
+void
+http_populate_cookies(struct http_request *req)
+{
+	int						 i, v, n;
+	struct http_cookie		*ck;
+	char					*c, *header;
+	char					*cookies[HTTP_MAX_COOKIES];
+	char					*pair[2];
+
+	if (!http_request_header(req, "cookie", &c))
+	   return;
+
+	header = kore_strdup(c);
+	v = kore_split_string(header, ";", cookies, HTTP_MAX_COOKIES);
+	for (i = 0; i < v; i++) {
+		for (c = cookies[i]; isspace(*c); c++)
+			;
+
+		n = kore_split_string(c, "=", pair, 3);
+		if (n != 2) {
+			kore_log(LOG_ERR, "malformed cookie: %s", c);
```

Comment:
I'd rather not have this log anything in case people are spamming requests with malformed cookies.